### PR TITLE
Add Adafruit_BME280 Library

### PR DESCRIPTION
**Pull request by mschader**
This will add the Adafruit_BME280 Library to Sming to have compatibility for the Sensor.

### CODE REVIEW

**Comment by slaff** [comment]
@mschader Can you add this library as submodule? Use something like this:

```
cd $SMING_HOME/..
git submodule add https://github.com/adafruit/Adafruit_BME280_Library.git \
Sming/Libraries/Adafruit_BME280_Library
```

And ONLY if there is a need for local changes then add those changes to Sming/Libraries/.patches/Adafruit_BME280_Library.patch.

**Comment by mschader** [comment]
@slaff Done so.

File: .gitmodules [hunk 1]
```diff
@@ -45,3 +45,6 @@
 	path = Sming/Libraries/Adafruit_SSD1306
 	url = https://github.com/adafruit/Adafruit_SSD1306.git
 	ignore = dirty
+[submodule "Sming/Libraries/Adafruit_BME280_Library"]
+	path = Sming/Libraries/Adafruit_BME280_Library
+	url = https://github.com/adafruit/Adafruit_BME280_Library.git
```
Comment (slaff) [inline .gitmodules]: @mschader Please, add also that option to the submodule:
`ignore = dirty`

File: Sming/Libraries/.patches/Adafruit_BME280_Library.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+-#include <Adafruit_Sensor.h>
+-#include <Wire.h>
++#include "Adafruit_Sensor.h"
++#include <../../SmingCore/Wire.h>
+ 
+ /*=========================================================================
+     I2C ADDRESS/BITS
+     -----------------------------------------------------------------------*/
+-    #define BME280_ADDRESS                (0x77)
++    #define BME280_ADDRESS                (0x76)
```
Comment (slaff) [inline Sming/Libraries/.patches/Adafruit_BME280_Library.patch]: @mschader I have no idea about this sensor but I am curious to know what is the reason for changing the address here?

File: Sming/Libraries/.patches/Adafruit_BME280_Library.patch [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+index 61aeeed..b9c3e7f 100644
+--- a/Adafruit_BME280.h
++++ b/Adafruit_BME280.h
+@@ -23,13 +23,13 @@
+  #include "WProgram.h"
+ #endif
+ 
+-#include <Adafruit_Sensor.h>
+-#include <Wire.h>
++#include "Adafruit_Sensor.h"
```
Comment (slaff) [inline Sming/Libraries/.patches/Adafruit_BME280_Library.patch]: Where is `Adafruit_Sensor.h` coming from? The compilation fails due to that missing file. It looks to me that also this `https://github.com/adafruit/Adafruit_Sensor` should be added as a submodule.

File: Sming/Libraries/.patches/Adafruit_BME280_Library.patch (hunk 2 above)
Comment (slaff) [inline Sming/Libraries/.patches/Adafruit_BME280_Library.patch]: Just for my understanding why is the address value changed here?

File: .gitmodules (hunk 1 above)
Comment (slaff) [inline .gitmodules]: @mschader Please, add also the following line to the file:
`ignore = dirty`

File: Sming/Libraries/.patches/Adafruit_BME280_Library.patch [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
++#include <../../SmingCore/Wire.h>
```
Comment (slaff) [inline Sming/Libraries/.patches/Adafruit_BME280_Library.patch]: Have you tried using <Wire.h> as in the original code?

File: Sming/Libraries/.patches/Adafruit_BME280_Library.patch (hunk 2 above)
Comment (mschader) [reply Sming/Libraries/.patches/Adafruit_BME280_Library.patch]: The I2C Address can differ between chip models. I bought several from different resellers and tested them and found none using address 0x77. Thats why i decided to change the default to 0x76.

**Pull request closed without merging**